scalyr_agent/monitor_utils/journald_checkpoints.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2014 Scalyr Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------
+# author:  Imron Alston <imron@scalyr.com>
+
+__author__ = 'imron@scalyr.com'
+
+import random
+import threading
+import scalyr_agent.util as scalyr_util
+import scalyr_agent.scalyr_logging as scalyr_logging
+
+global_log = scalyr_logging.getLogger(__name__)
```

Comment:
Why have this as a separate file?

Reply:
Not for anything relevant to this plugin, but as I was writing it, I thought that this is not the only plugin that requires saving checkpoints (docker plugin does it also) and future plugins will probably need this functionality at some point too, so my thinking was it might make sense to have a generic way of handling monitor checkpoints and ended up putting this in its own file in case it gets expanded for other plugins later.

Reply:
I hear you about wanting to reuse it on other plugins.  However, until we actually have multiple plugins using it, I suggest we don't put it in its own file.  Essentially, don't try to generalize until we actually have multiple uses.